src/commands/pool.rs

```diff
@@ -14,20 +16,31 @@ pub enum PoolCommand {
     Open(String, // name
          Option<String>, // config
          Box<Fn(Result<i32, PoolError>) + Send>),
+    OpenAck(i32, // cmd id
+            Result<i32 /* pool handle */, PoolError>),
     Close(i32, // pool handle
           Box<Fn(Result<(), PoolError>) + Send>),
+    //TODO unify type of result for open/close/refresh? (Fn(Result<i32, PoolError>))
+    CloseAck(i32,
+             Result<(), PoolError>),
     Refresh(i32, // pool handle
-            Box<Fn(Result<(), PoolError>) + Send>)
+            Box<Fn(Result<(), PoolError>) + Send>),
+    RefreshAck(i32,
+               Result<(), PoolError>),
 }
 
 pub struct PoolCommandExecutor {
-    pool_service: Rc<PoolService>
+    pool_service: Rc<PoolService>,
+    pending_cbs_opening: RefCell<HashMap<i32, Box<Fn(Result<i32, PoolError>)>>>,
```

Comment:
I suggest to use separate callbacks queues:
* open_callbacks
* refresh_callbacks
* close_callbacks